src/components/Hotspot.tsx

```diff
@@ -148,6 +151,43 @@ export default class Hotspot extends Component<Props, State> {
     }
   };
 
+  getFontSize = (layout: any, hotspot: any, label: string): number => {
+    let textEl = document.createElement('div');
```

Comment:
let's use Preact approach:
we can render div (without text, f.e. use 0 as default fontSize), get refs from it. Then once component did mount - make your calculation and set fontSize value to state.